src/backend/src/yaksuri_progress.c

```diff
@@ -1065,333 +1065,439 @@ static int set_multichunk_subreq(const void *inbuf, void *outbuf, uintptr_t coun
     return rc;
 }
 
-int yaksuri_progress_enqueue(const void *inbuf, void *outbuf, uintptr_t count, yaksi_type_s * type,
-                             yaksi_info_s * info, yaksa_op_t op, yaksi_request_s * request)
+/* Subroutines to setup subreq for pack with different buffer types */
+static int set_pack_d2_subreq(const void *inbuf, void *outbuf, uintptr_t count, yaksi_type_s * type,
+                              yaksi_info_s * info, yaksa_op_t op, yaksi_request_s * request,
+                              yaksuri_request_s * reqpriv, uintptr_t threshold,
+                              yaksuri_subreq_s * subreq)
```

Comment:
`d` means "device", right? What does `2` refer to?

Reply:
Yes, `d` means `device`. `2` refers to `to`. It tries to follow the naming of existing functions such as `pack_d2d_nop2p_acquire`

Reply:
I see. How about `set_subreq_pack_from_device`?

Reply:
The proposed format may make some subroutine names very long, e.g.,`set_subreq_pack_from_registered_host`, `set_subreq_pack_from_device_to_device`, `set_subreq_pack_from_device_to_registered_host`. I think it is fine to use these abbreviation consistently in this file.

If you are OK with the abbreviation, I am not sure which is better between `set_pack_d2_subreq ` and `set_subreq_pack_d2`. What do you think?

Reply:
We are more used to `d2d`, `d2h`, etc. although a short comment above the definition to expand the abbreviation can be helpful.

`_d2` is just too cryptic, mostly because we don't use such often. When used, it often means different from the meaning here. One of the heuristics (from "Clean code") is the name needs to be pronounceable and can be put into a sentence. `d2` doesn't make sense in a sentence.

Reply:
Agreed with your comment. But just the suggested name seems too long and a bit inconsistent with the other subroutines which use `d2d, d2h`. I guess the confusing part is we don't have the "destination" in `d2`. Will `d2any` read better?

Reply:
`from_dev` is 8 characters. `d2any` is 5 characters. Are we bargaining here? :)

Reply:
How about `from_registered_host` and `from_managed`? I am more concerning that it is not consistent with the other subroutines' name.

Reply:
Now you mentioned, were you using `m` to encode `managed`? It was very cryptic to me. The `from_xxx` routines are at one higher level than `_x2y` routines, I think the different name patterns are ok here -- the hierarchy is more important to read than the individuals. As for the two names, how about `from_reghost` (I'll accept `from_rhost` with some reluctance)? `from_managed` seems to be a minimal readable name to me.

Reply:
`from_rhost` seems good to me.